### DISCUSSION TITLE
Capacitor recommendations for LYWSD03MMC?

### DISCUSSION THREAD

**Discussion by marvinlehmann** [pvvx/ATC_MiThermometer#511] [Q&A]
I often read about adding capacitors to C24 & C25 to improve stability and operational time. I went through many comments and collected information as I have little understanding of electronics unfortunately.

- operating voltage: up to ~3.1V (?)
- average power consumption: 14-15 μA (https://pvvx.github.io/ATC_MiThermometer/CustPower.html)
- capacitor:
   - higher voltage rating -> lower leakage current
   - larger capacitance -> greater leakage current

recommendation:
- _"10 μF"_ (https://github.com/pvvx/ATC_MiThermometer/issues/265#issuecomment-1723248115)
  - _"cheap ones, with capacities of more than 20 UF, have a leak comparable to the consumption of the thermometer itself"_

- _"4.7..47 μF"_ (https://github.com/pvvx/ZigbeeTLc/issues/37#issuecomment-1870491246)

- _"Desirable more than 10 μF (С24) and more 100 nF (C25)"_ (https://github.com/pvvx/ATC_MiThermometer/issues/172#issuecomment-1007169665)

- only one capacitor is needed as C24 & C25 are in parallel (https://github.com/pvvx/ATC_MiThermometer/issues/11#issuecomment-870230398)

- 100 nF & 2,2 μF  (http://forum.jdtech.pl/Watek-mijia-lywsd03mmc-termohigrometr-z-wyswietlaczem-lcd-ble)

- formula for dimensioning the capacitor... (https://www.ti.com/lit/wp/swra349/swra349.pdf)

So... you people who did the upgrade.. which capacitor[s] did you get? I don't really know what to look for. :'-)

EDIT:
I just noticed that the 2.0 Board doesn't seem to have a C24 (?) so a small cap for C25 is necessary.
I also wondered what the voltage should be used because of bias voltage and leakage. At least 6.3V I guess. But what about something like 50V...?

Contrary to the earlier statement:

> This suggests that, if low leakage current is of importance, then a lower voltage rated capacitor is preferred.

page 10 of [Current Leakage through Ceramic Capacitors](https://www.silabs.com/documents/public/application-notes/an1415-current_leakage_through_ceramic_capacitors.pdf)

**Comment by tido-** [comment]
\* WIP *
@marvinlehmann, thank you. I followed your links.
 
**Bluetooth power consumption on data transfer**
RX TX +0 dBm -> Maximum pulse current of the entire device: 6 mA
RX TX +10 dBm -> Maximum pulse current: less than 25 mA

All Qingping devices - all capacitors are installed.
All Xiaomi devices - no capacitors installed :)
https://github.com/pvvx/ATC_MiThermometer/issues/172#issuecomment-1007169665

**SMD size:**
c24 1206 	3,20 mm
c25 0603 	1,60 mm
Desirable more than 10 μF (С24) and more 100 nF (0,1uF) (C25).

MLCC (ceramic) does not have polarity and you should choose it in most cases. However, if you decided to use a tantalum capacitor, or any other type of capacitor that has a polarity, DO NOT REVERSE THEM! Next to the "C24" or "C25" sign is the positive pole.

So, which technic or better which **type**&**manufacturer** leaks less - as we try to save power?
https://github.com/pvvx/ATC_MiThermometer/issues/265#issuecomment-1761777750

---
Leakage current (DCL) is common effect of all capacitors and its value and dependencies on electrical and environmental conditions are related to capacitor technology.
See "MAX. DCL" in the documentation for capacitors  or "Insulation resistance"...

vishay - Solid Tantalum Surface Mount Chip Capacitors, TANTAMOUNTTM, Molded Case, **Very Low DCL**
Capacitance Range: 0.1 μF to 470 μF  |  Voltage Rating: 4 V to 50 V

c24 1206 3,20 mm 10 μF 

TYPE: TL3
CASE CODE: A
CAPACITANCE: 10 μF
DC VOLTAGE: 10 V
MAX. DCL: 0.5 μA 
MAX. DF: 6
MAX. ESR: 3.400 \ 2000

Potential Partnumbers:
search term: capacitor 1206 10uF 10V SMD
TL3 A 106 010 3400  search term: (vishay TL3A106(1)010(2)3400)
TL3 A 106 010 2000  search term: (vishay TL3A106(1)010(2)2000)

22 μF TL3A226(1)6R3(2)2900  0.66 μA
33 μF TL3A336(1)004(2)2900  0.66 μA

\* WIP *

**Reply by danieldur** [reply]
I am more or less, electronically challenged, but also curios as to why things are in a certain way so, I feel forced to ask a couple of questions.
tido-, towards the end of the above mentioned white paper (https://www.ti.com/lit/wp/swra349/swra349.pdf) the calculated capacitor size is about 87 - 100 uF. Given this, is there any reason why you recommend values that are around 10 - 15 % of that? Is it to make the ideal compromise between the capacitor size and leakage current, and 10, 22 or 33 uF are big enough for the job while not discharging the battery too much?
Thank you very much!

**Reply by marvinlehmann** [reply]
I'm still wondering the same.
And as a noob I tried the formula like:
```
(8mA * 100ms + 6µA * 2400ms - (2.2V / 2200Ω) * 2500ms) / (2.7V - 2.2V) = -3.4pF ???
```

I based it on the values from [CustPower](https://pvvx.github.io/ATC_MiThermometer/CustPower.html), https://github.com/pvvx/ATC_MiThermometer/issues/477#issuecomment-1965376436 and I measured the 2200Ω myself at the battery slot contacts because I just don't know any better x)


If I base it on the graph from https://github.com/pvvx/ATC_MiThermometer/issues/477#issuecomment-1965735927 and also omit the sleep load:

```
C = (4mA * 3ms + 6mA * 2ms + 8mA * 1ms - (2.2V / 2200Ω) * (3+2+1)ms) / (2.7V - 2.2V) = 52µF

t = 2200Ω * 52µF * ln((2.7V - 2.2V) / (2.7V - 2.6V)) = 0.2s
```
**But again, I don't really know what I'm doing..**

I also don't understand why there are different capacitance recommendations for C24 & C25 when the capacitors "slots" are in parallel? Not that it would be relevant for 2.0 boards that only have C25.

**Reply by pvvx** [reply]
Take these values: average current of 5 mA and activity period of up to 11 ms with transmission. This will be optimal for average operating conditions of current FW in beacon mode.
The calculation formula is specified in [this document](https://www.ti.com/lit/wp/swra349/swra349.pdf).
For connection modes, especially with active pin code, a capacitor that is too large will be required and this should not be taken into account...

> I also don't understand why there are different capacitance recommendations for C24 & C25 when the capacitors "slots" are in parallel?

A capacitor with a large capacity has a large [ESR](https://en.wikipedia.org/wiki/Equivalent_series_resistance). Therefore, it is shunted by a small capacitor with a small ESR.

**Reply by marvinlehmann** [reply]
> Take these values: average current of 5 mA and activity period of up to 11 ms with transmission.

So `C = (5mA * 11ms - (2.2V / 2200Ω) * 11ms) / (2.6V - 2.2V) = 110µF` ?
Unsure about Ri / internal resistance though. When using 1000Ω it would be 77µF.

`t = 2200Ω * 110µF * ln((2.7V - 2.2V) / (2.7V - 2.6V)) = 0.4s`  -> as the paper also states 

> If t is considerably shorter than the sleep state, the capacitor is unnecessary large and can be reduced by increasing Vmax.

So I could set Vmax from 2.6V to like 2.69999V 
```
C = (5mA * 11ms - (2.2V / 2200Ω) * 11ms) / (2.69999V - 2.2V) = 88µF
t = 2200Ω * 88µF * ln((2.7V - 2.2V) / (2.7V - 2.69999V)) = 2s
```

> A capacitor with a large capacity has a large [ESR](https://en.wikipedia.org/wiki/Equivalent_series_resistance). Therefore, it is shunted by a small capacitor with a small ESR.

Ahhhh okay.  Thank you!

LATE EDIT: I just noticed after rereading the document that I misinterpreted Ri as the resistance of the circuit the battery should be able to manage. So the 2200Ω are complete non-sense x)

**Reply by pvvx** [reply]
With an internal battery resistance of 200 ohms, a typical thermometer no longer works.
The battery voltage drop when the transmitter is operating at 0 dBm will be 200*0.007A = 1.4V.
When the internal resistance of the battery approaches 100 ohms, the battery will only last a few days.

**Reply by pvvx** [reply]
I'm sure 47..68 µF will be quite enough. With this you will already face another problem - when replacing the battery you will have to wait several minutes until this capacitor discharges.
And subsequent switching on will be difficult.
When the voltage is below 2.2V in the thermometer, all chips are switched to deep sleep mode for 3 minutes. When starting, the battery needs to charge this capacitor. The charge goes smoothly, and the thermometer has time to start and understand that the voltage is below 2.2V. And it falls asleep in the hope that after 3 minutes everything will be normal and you can start working...
For this reason, large capacities are not recommended.
And 22 μF will also be sufficient.

**Comment by pvvx** [comment]
> Bluetooth power consumption on data transfer xxx mA?

Default config: RX TX +0 dBm -> Maximum pulse current of the entire device: 6 mA
+10 dBm -> Maximum pulse current: less than 25 mA 

The internal resistance when the CR2032 battery is not completely discharged is more than 100 Ohms. Therefore, with a current of 6 mA CR2032, the battery will produce 3.0 - 100 * 0.006 = 2.4 V (+25С).
The capacitor smoothes the pulse and the voltage does not reach the lower operating voltage of the circuit.

**Comment by AleXSR700** [comment]
So did anybody ever solder in a capacitor at C24 or C25? If so, any measurements for different capacitors?

Still thinking about whether to buy some, e.g. from AliExpress: 
https://a.aliexpress.com/_EQ1mZSh

e.g. A 35V1UF 105

**Reply by tido-** [reply]
Not yet. I was to lazy to search on AliE to find a non-cheating real brand **Very Low DCL** capacitor. 
If you buy cheap, you buy twice.  Is an old saying.
So, I was considering a shop in Europe like Distrelec, Mouser, Farnell or such.

I found promising one for c24 based on the hints of @pvvx. If somebody does the same for the c25, I would start to look at resellers.

**Reply by AleXSR700** [reply]
I thought only one was needed. Either C24 or C25. They are parallel, I thought.

**Reply by tido-** [reply]
I guess this is just like a car with TwinTurbo charger (Toyota Supra (1993–2002)). The small is quicker and the large supports greater load. And yes they are in parallel to my understanding too.

**Comment by danieldur** [comment]
So, in the end, is there any clear conclusion as to what values should be used? Could we maybe use the same capacitors as what Qingping puts in their devices?
Thanks!

**Reply by pvvx** [reply]
The calculation of capacity is shown in [this document](https://github.com/user-attachments/files/18102754/swra349.pdf).
But you shouldn't pay so much attention to cheap things.

**Comment by agittins** [comment]
For anyone who might be interested in some historical stats, here's some from my setup....
<img width="1649" height="408" alt="image" src="https://github.com/user-attachments/assets/524c1215-58a3-48a6-8139-f0b3ac509e2b" />

- Xiaomi LYWSD03MMC units hardware version B1.7
- All configured with 2.5 second advertisements, BTHomeV2, encrypted (variation possible due to user-error, but pretty confident)
- 7 or 8 ESPHome bluetooth active proxies, mostly on esp32 classic. Mostly 320 interval, 300ms window, active, using HA `api` (not mqtt)

Some notes from the above:
- Green trace - Installing a 47uF 10v Electrolytic capacitor looks really interesting - we are at 17 months and the battery still reads at over 50%. The initial drop in capacity I think is from the "forming" process - where the capacitor has a high leakage until it has stabilised at a given voltage for a length of time. It looks like this took a week or two, and cost nearly 10%. I think the next time I do this I'll "pre-form" the caps using an external supply before putting them into the battery-powered device.
- Orange trace - Installing a 10uF "monolithic" ceramic seems to be following a similar discharge curve to the unmodified devices, even though its cell is 2.5 months younger. If you offset the curve to account for that, it falls somewhere between the pink and the yellow/blue traces, so maybe it made things slightly worse. However we might find that the tail-end of the discharge will be longer for the capped device, time will tell. Worth noting this is in a west-facing bathroom that is less-used than the ensuite (mustard) but gets wide temp swings and occasional wide humidity swings - so maybe the cap is helping more than is obvious from the data so far. **This device also has a reed switch installed (almost always open) which may have a marginal affect as well**.
- The 2024-10-31 batteries (pink, blue, mustard, brown) all look like around an 18-month lifespan, give or take a few months.
  - The pink one which has the shortest life of those is outside, and is exposed to generally higher humidities and slightly lower temps (we are in a very temperate climate, so it rarely goes below 10C). My guess is that the higher humidity and perhaps the slightly wider temp swings leads to lower lifespan.
  - The brown one is located next to our ducted AC's thermostat, so in theory it probably has the smallest temp and humidity swings, which *might* explain its stronger performance.
  - The mustard trace is in an ensuite that experiences high humidity swings due to showers etc.
  - The blue trace is in an uninsulated garage that also has about a kW of server heat, so experiences generally higher temps than internal *or* ambient, but lower swings.
- Purple trace - this no-name cell is looking on track to last maybe 8 months, but hard to be sure since the discharge curve is tailing off a bit. This looks consistent with a cell that has higher internal resistance, which is probably to be expected for a no-name cell.

Some opinionated conclusions...

- Brand (or perhaps, supply-chain) matters, but maybe this is more about internal resistance than capacity. Buying cells from a local retail outlet with traditional supply chain seems to be more reliable - but at a steep cost. It would be interesting to see if a large reservoir cap might smooth that out for longer.
- A "big" (47uF) electro cap does look like it's making a big difference - but it's too soon to tell for sure, yet. I used a no-name electro from an AliExpress assortment pack(!), so I am moderately impressed with the result so far.
- Smaller 4.7uF ceramic is less impressive, possibly harmful - but still too early to tell, as it seems to be showing signs of a long-tail.
- Large temp and/*or* humidity swings might be detrimental to cell longevity, but probably only to a few months out of 18. Note that the conditions were never below 10C. Not shown is a device in our barfridge, installed late 2025. It dropped to about 70% within a week or two, and is only now getting down to 60% at 8months - voltage depression at lower temps is a normal aspect of these cells so probably not inherently meaningful.

Some opinionated recommendations:
- Buy name-brand cells from a large local retailer (or someone who has a reputation at risk, like DigiKey, Farnell etc). Off-brand cells are unpredictable, and counterfeits are common.
- Capacitance on the power rail appears to help. In order of importance, I would:
- Add a large Electrolytic cap of around 47uF. Keep the voltage spec as low as possible.
- If using cheap or old electrolytic caps, pre-form that capacitor by connecting it to an externally powered 3v rail until it stops leaking current. Then come back here and post stats about whether that had any effect! It's probably better though to use tantalum or ceramic caps though for their lower leakage figures. Ceramic caps have to be derated a fair bit due to their DC bias effect (where there effective capacitance is reduced relative to how close to their rated voltage they are held at).

So not a very solidly-scientific set of conclusions, and my data is still incomplete but thought I'd throw it in here while I had the graph loaded up :-)

**Reply by pvvx** [reply]
CR2050
<img width="640" height="330" alt="image" src="https://github.com/user-attachments/assets/610978c1-3985-415b-80b0-0f143aac5330" />
